lib/driver.js

```diff
@@ -225,12 +226,12 @@ class XCUITestDriver extends BaseDriver {
       device.setScaleFactor(this.opts.scaleFactor);
     }
 
+    await this.runReset();
+
     // handle logging
     this.logs = {};
     await this.startLogCapture();
 
-    await this.runReset();
```

Comment:
I suppose where we really want the reset is at the beginning of the session---ensure that a new test starts in the right state. We could also do both but that's kind of a waste of time, I agree.